java/src/jmri/jmrit/vsdecoder/Diesel3Sound.java

```diff
@@ -663,7 +663,9 @@ public void run() {
                 }
                 _sound.stop();
             } catch (InterruptedException ie) {
-                log.error("execption", ie);
+                // kill thread
+                log.debug("thread interrupted");
+                return;
```

Comment:
This return is a `no operation`, meaning that it doesn't do anything useful. The catch statement is outside of the `while` loop so if an exception occurs, it will abort the `while` loop and exit the run() method even without the `return` statement.

Reply:
Thanks for the feedback.
The return was introduced in a thread similar to this, see https://github.com/JMRI/JMRI/pull/10290 . As far as I understand the reason was to "kill the threads left running at end of tests". I am open to change this PR as you suggest.

Reply:
> The return was introduced in a thread similar to this, see #10290 . As far as I understand the reason was to "kill the threads left running at end of tests". I am open to change this PR as you suggest.

Yes, it was introduced in that PR. But the return doesn't do anything there either. If an InterruptedException occurs, the loop will be aborted and the method is exited. That's because the while loop is inside of the try-catch statement.

Which problem are you trying to solve?

Reply:
> Which problem are you trying to solve?

```Diesel3Sound.java``` and ```Steam1Sound.java``` have a similar thread case. One has a ```return```, the other not. My intention was a cleanup.

Reply:
Alright.  Will do so.  Thanks!